class-auth.php

```diff
@@ -100,9 +100,25 @@ public function register_rest_routes() {
 	 * Add CORs suppot to the request.
 	 */
 	public function add_cors_support() {
+		global $wp_version;
+
 		$enable_cors = defined( 'JWT_AUTH_CORS_ENABLE' ) ? JWT_AUTH_CORS_ENABLE : false;
 
-		if ( $enable_cors && !headers_sent() ) {
+		if ( ! $enable_cors ) {
+			return;
+		}
+
+		// Hook exists since 5.5.0
+		if ( version_compare( $wp_version, '5.5.0', '>=' ) ) {
+			add_filter( 'rest_allowed_cors_headers', function ( array $headers ) {
+
+				$filters = apply_filters( 'jwt_auth_cors_allow_headers', 'X-Requested-With, Content-Type, Accept, Origin, Authorization, Cookie' );
+
+				$split = preg_split( "/[\s,]+/", $filters );
+
+				return array_merge( $headers, $split );
```

Comment:
Indeed, https://developer.wordpress.org/reference/classes/wp_rest_server/serve_request/ does not make the list unique after filtering, so we need to do it manually ourselves.
```suggestion
				return array_unique( array_merge( $headers, $split ) );
```